registry/registry.go

```diff
@@ -123,6 +125,23 @@ func (r *RouteRegistry) register(uri route.Uri, endpoint *route.Endpoint) route.
 	}
 
 	endpointAdded := pool.Put(endpoint)
+	//Check endpoint for a load balancing algorithm. If it does exist & differs from that of the pool, then take the lb algorithm of the endpoint, otherwise keep the pools lb algorithm.
+	if len(endpoint.LoadBalancingAlgorithm) > 0 && strings.Compare(endpoint.LoadBalancingAlgorithm, pool.LBAlgorithm) != 0 {
+		if config.IsLoadBalancingAlgorithmValid(endpoint.LoadBalancingAlgorithm) {
+			pool.Lock()
```

Comment:
The locking needs some improvement, in the previous if statement you already read from the pool without holding a lock.

Reply:
@maxmoehl The locking was intentionally done only for write operations, to avoid unnecessary locks for a frequent operation with only a reading access to the pool. Could you provide an example, where this lock strategy could lead to a problem or an inconsistent state, except that we only overwrite a value, which was currently already changed? The result would be correct in this case though. @peanball For the case, that you a bored in your vacation and also want to discuss :)

Reply:
You'll still need a read lock to avoid inconsistent read during write. This can be at field or even word level. 
The write is not guaranteed to be atomic, so reads need to have locks too that are exclusive with write locks (and vice versa).

Reply:
The correct lock strategy is commited and I am resolving this conversation. Thanks!